### ISSUE TITLE
¿Cómo regresamos un 200 OK?

### ISSUE THREAD

**Issue by defvol** [mxabierto/api-standards#5]
No necesariamente será un array de results con paginación.

¿Qué hacemos en estos casos?

**Comment by defvol** [comment]
Siguiendo http://jsonapi.org/format/

se me ocurre:

``` json
{
    "results": [
        "OK"
    ],
    "errors": []
}
```

**Comment by defvol** [comment]
y para error:

``` json
{
    "results": [],
    "errors": [
        {
            "message": "Descripción del error.",
            "exception": "[stacktrace detallado]"
        }
    ]
}
```

**Comment by defvol** [comment]
Las dudas surgieron en https://github.com/mxabierto/atencion-ciudadana/wiki/API-del-Sistema-de-Atenci%C3%B3n-Ciudadana

**Comment by chuckcfs** [comment]
@rodowi yo diria que depende del método, si es un POST, PUT o DELETE para con id el resultado debería ser el objeto creado, actualizado, borrado o leído:

``` json
{
  "id" : 13141,
  ...
}
```

si fuera un GET para un listado de datos ahí si deberíamos usar el campo results

``` json
{
  "results": [],
  "pagination": {}
}
```

y si hubiera un error se envia el código 4xx o 5xx y el resultado debería ser sólo el objeto de error:

``` json
{
  "message" : "Descripción del error.",
  "exception": "[stacktrace detallado]"
}
```

**Comment by carlosmaya** [comment]
coincido con @chuckcfs , me hace sentido que regrese el "id" en el caso del POST, PUT y DELETE

me queda la duda con el GET, ¿cuál es un escenario en donde hago un GET sin `results` ? eso me suena a un error, el "resource" no existe

**Comment by chuckcfs** [comment]
no, más bien yo no digo regresar solo el id, sino el objeto completo, el objeto que se cree, actualice, borre o lea

@carlosmaya el GET me refiero a:
- Objeto - ej. GET articles/1211

``` json
{
  "id" : 121313,
  "title" : "Lorem ipsum",
  "content" : "Lorem ipsum dolor sit amet...",
  ...
}
```
- Lista - ej. GET articles/

``` json
{
  "results" : [
    {
      "id" : 121313,
      "title" : "Lorem ipsum",
      "content" : "Lorem ipsum dolor sit amet...",
      ...
    }
  ],
 "pagination" : {
    "page" : 1,
    "per_page" : 20,
    "total" : 1
  }
}
```

**Comment by carlosmaya** [comment]
@chuckcfs  got it!! 

@rodowi viendo la wiki del API, no me queda claro ¿por qué  en las respuestas de éxito también se regresa `errors` y lo mismo para `results` en la de Error? solo para entender

**Comment by chuckcfs** [comment]
no, creo que ahí esta la confusión, en los resultados existosos no se debe regresar "errors", sólo en los que algo sale mal

**Comment by captain-http** [comment]
también me agrada la propuesta de @chuckcfs

**Comment by defvol** [comment]
de acuerdo con @chuckcfs, por lo que veo lo que se acordó fue (me corrigen si me equivoco):
- errors y results no deben aparecer en la misma respuesta
- results sólo se regresa cuando estamos accesando listas, p. ej. GET /notifications, /comments
- results siempre va acompañado de pagination
- en POST, PUT, DELETE, GET con id, se regresa el objeto en cuestión

**Comment by defvol** [comment]
hey @chuckcfs lo quieres agregar a la guía?

**Comment by defvol** [comment]
ya actualicé lo que acordamos en:
https://github.com/mxabierto/sara/wiki/API-del-Sistema-de-Atenci%C3%B3n-Ciudadana
https://github.com/mxabierto/sara/wiki/API-del-Clasificador
